src/nvim/terminal.c

```diff
@@ -465,12 +453,8 @@ static int terminal_execute(VimState *state, int key)
       break;
 
     case K_EVENT:
-      // We cannot let an event free the terminal yet. It is still needed.
-      s->term->refcount++;
       multiqueue_process_events(main_loop.events);
-      s->term->refcount--;
```

Comment:
Sounds good. In the meantime, don't want to overwhelm  @madmax28 with side-jobs. The ASan error reported by @oni-link below is likely related to this, so `refcount` should stay until we make the change you suggest.